inference-engine/thirdparty/clDNN/src/space_to_batch.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "space_to_batch_inst.h"
+
+#include "primitive_type_base.h"
+#include "error_handler.h"
+#include "json_object.h"
+#include "data_inst.h"
+#include <string>
+#include <vector>
+
+namespace cldnn {
+primitive_type_id cldnn::space_to_batch::type_id() {
+    static primitive_type_base<space_to_batch> instance;
+    return &instance;
+}
+
+layout space_to_batch_inst::calc_output_layout(space_to_batch_node const& node) {
+    auto desc = node.get_primitive();
+
+    auto input_layout = node.input(0).get_output_layout();
+    auto input_format = input_layout.format;
+
+    const size_t dims_num = format::dimension(input_format);
+
+    // Getting data from constant inputs. There are 3 args: block_shape, pads_begin, pads_end
+    std::vector<std::vector<int32_t>> args;
+    for (size_t i = 1; i < node.get_dependencies().size(); ++i) {
```

Comment:
Since we don't need these inputs in runtime, I propose to read the values from them inside the plugin and pass cldnn::tensor parameters instead of additional inputs to the primitive.